packages/legend-extension-external-language-morphir/src/ELMorphir_GenerationPlugin.tsx

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import packageJson from '../package.json';
+
+import type {
+  DSL_StudioPlugin_Extension,
+  GenerationFile,
+  FileGenerationState,
+  StudioPluginManager,
+  FileGenerationResultViewerAction,
+  FileGenerationScopeFilter,
+} from '@finos/legend-studio';
+
+import { StudioPlugin } from '@finos/legend-studio';
+import { NetworkClient, assertNonEmptyString } from '@finos/legend-shared';
+import type { PackageableElement } from '@finos/legend-graph';
+import { ConcreteFunctionDefinition } from '@finos/legend-graph';
+
+const MORPHIR_TYPE_NAME = `morphir`;
+
+interface ELMorphir_GenerationPluginConfigData {
+  morphirVisualizerUrl: string;
+  linterServerUrl: string;
+  linterAppUrl: string;
+}
+
+export class ELMorphir_GenerationPlugin
+  extends StudioPlugin
+  implements DSL_StudioPlugin_Extension
+{
+  networkClient: NetworkClient;
+  morphirVisualizerUrl = ``;
```

Comment:
the more idiomatic way to do this is to have some private flag `_morphirVisualizerUrl?: string | undefined` and have a getter that asserts these are set, e.g. `get morphirVisualizerUrl(): string { guaranteeNonEmptyString(...) }` that way, typigns are correctly done, but consumers get proper feedbacks when the plugins has not been configured, instead of getting empty strings

Reply:
Fixed -- using private fields with getters now.